### PR TITLE
[Draft][circle-execution-plan] Add Pal to ExecutionPlanner

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr adds pal interface for different platforms to correctly calculation of the size of the scratchpad.

ONE-DCO-1.0-Signed-off-by: Artem Balyshev a.balyshev@partner.samsung.com

### CODE REVIEW

**Comment by binarman** [comment]
I think this could be done by option, not PAL.
We used PAL in luci-interpreter, because:
1. It is impossible to actually build some pieces of code for all platforms (for example pthreads for MCU or ARM intrinsics for x86)
2. We wanted to save build time and runtime resources.

File: compiler/circle-execution-plan/src/CircleExecutionPlan.cpp [hunk 1]
```diff
@@ -35,6 +35,18 @@ int entry(int argc, char **argv)
...
+    .type(arser::DataType::STR)
+    .required(false)
+    .default_value("linux")
+    .help("Platform name: linux mcu cmsisnn");
+  arser.add_argument("--use_dsp")
+    .nargs(1)
+    .type(arser::DataType::BOOL)
+    .required(false)
+    .default_value(false)
+    .help("Plan with or without dsp");
```
Comment (binarman) [inline compiler/circle-execution-plan/src/CircleExecutionPlan.cpp:49]: I think it worth to mention that it applied to cmsis only for now.

File: compiler/circle-execution-plan/src/CircleExecutionPlan.cpp [hunk 2]
```diff
@@ -82,7 +115,7 @@ int entry(int argc, char **argv)
   auto module = importer.importModule(circle_model);
 
   // Do main job
-  circle_planner::ExecutionPlanner execution_planner(module->graph());
+  circle_planner::ExecutionPlanner execution_planner(module->graph(), platform_type, use_dsp);
```
Comment (binarman) [inline compiler/circle-execution-plan/src/CircleExecutionPlan.cpp:118]: What about make "TargetPlatform" structure/class and add `platform_type` and `use_dsp` there?
I think at some point we will need even more options, so this function definition will grow...

**Review by binarman — COMMENTED** [review]
Overall looks good to me

File: compiler/circle-execution-plan/src/CircleExecutionPlan.cpp (hunk 1 above)
Comment (BalyshevArtem) [reply compiler/circle-execution-plan/src/CircleExecutionPlan.cpp]: Done

File: compiler/circle-execution-plan/CMakeLists.txt [hunk 3]
```diff
@@ -1,16 +1,3 @@
-set(SOURCES
```
Comment (binarman) [inline compiler/circle-execution-plan/CMakeLists.txt:1]: Maybe merge this and cmake file from src dir?

Comment (BalyshevArtem) [reply compiler/circle-execution-plan/CMakeLists.txt:1]: Did I understand you correctly, that you are suggesting to use only this (`compiler/circle-execution-plan/CMakeLists.txt`) cmakeList and don't create new `compiler/circle-execution-plan/src/CMakeLists.txt`?

**Comment by BalyshevArtem** [comment]
everything has been added, close this draft

**Pull request closed without merging**